### PR TITLE
Fix `devtools::check_built failure` failure

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Fix `devtools::check_built failure` failure caused by the latest MLflow R release:

https://github.com/mlflow/mlflow/runs/4175272472?check_suite_focus=true#step:14:138

```
── R CMD check results ────────────────────────────────────── mlflow 1.21.1 ────
Duration: 52.7s

❯ checking CRAN incoming feasibility ... NOTE
  Maintainer: ‘Matei Zaharia <matei@databricks.com>’
  
  Days since last update: 0

0 errors ✔ | 0 warnings ✔ | 1 note ✖
```

## How is this patch tested?

Make sure the R check passes.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/.run-tests.R
Comment (harupy) [inline mlflow/R/mlflow/.run-tests.R:21]: This is hacky but I could not find another way.

File: mlflow/R/mlflow/.run-tests.R
Comment (liangz1) [inline mlflow/R/mlflow/.run-tests.R:7]: How about we just do `error_on="warning"`.
According to https://devtools.r-lib.org/reference/check.html
```
Passing R CMD check is essential if you want to submit your package to CRAN: you must not have any ERRORs or WARNINGs, and you want to ensure that there are as few NOTEs as possible.
```
So it's acceptable to have one or more notes. According to  https://stackoverflow.com/a/23831508/12165968, the maintainer note is safe to ignore.

The downside of my proposal is that we might be less alerted if other notes show up.

Comment (harupy) [reply mlflow/R/mlflow/.run-tests.R:7]: In mlflow 1.21.0, one note (about an invalid URL) blocked the package **publication**.

Comment (liangz1) [reply mlflow/R/mlflow/.run-tests.R:7]: Got it. It's frustrating that CRAN also categorizes blocking issues as a note besides non-blocking ones...
Do we know whether disabling the CRAN incoming feasibility check will miss blocking issues? I didn't understand what exactly CRAN incoming feasibility check does and why setting the 7 days threshold resolves the issue...

Comment (harupy) [reply mlflow/R/mlflow/.run-tests.R:7]: > Do we know whether disabling the CRAN incoming feasibility check will miss blocking issues?

> 


> I didn't understand what exactly CRAN incoming feasibility check does.

It performs a bunch of checks. You can take a look at this file:
https://github.com/wch/r-source/blob/4561aea946a75425ddcc8869cdb129ed5e27af97/src/library/tools/R/QC.R#L8005-L8008


> why setting the 7 days threshold resolves the issue.

```
── R CMD check results ────────────────────────────────────── mlflow 1.21.1 ────
Duration: 52.7s

❯ checking CRAN incoming feasibility ... NOTE
  Maintainer: ‘Matei Zaharia <matei@databricks.com>’
  
  Days since last update: 0

0 errors ✔ | 0 warnings ✔ | 1 note ✖
```

This note basically means `it's been 0 days since the last release. It's too early`. The code I attached indicates that the threshold is 7 days. It looks like there is no option to disable this recency check so we just disable the entire feasibility check by setting `remote` to `FALSE`.

Comment (liangz1) [reply mlflow/R/mlflow/.run-tests.R:7]: I see. I previously thought `checking CRAN incoming feasibility ... NOTE
  Maintainer: ‘Matei Zaharia <matei@databricks.com>’` just means a complaint about the maintainer.

So, as long as we don't release two versions within 7 days, it would run through the same checks as it was before this PR.

Comment (harupy) [reply mlflow/R/mlflow/.run-tests.R:7]: Yep :)

**Review by liangz1 — APPROVED** [review]
Thanks for fixing this issue! (I'm not very confident about this PR so you can decide whether you want another reviewer to take a look.)

**Comment by harupy** [comment]
Thanks for the review and comments! To be honest, I'm not 100% confident about the 7 days threshold but I'll merge the PR and see what happens in 7 days :)

**Pull request merged**
